driver-core/src/test/unit/com/mongodb/AuthConnectionStringTest.java

```diff
@@ -104,9 +99,21 @@ private void testValidUris() {
             }
 
             assertMechanism("credential.mechanism", credential.getMechanism());
+        } else {
+            if (!getExpectedValue("credential").equals(BsonNull.VALUE)) {
+                fail(String.format("Connection string '%s' should not produce credentials", input));
+            }
         }
     }
 
+    @Nullable
+    private MongoCredential getMongoCredential() {
```

Comment:
Not sure this adds any value given that 1 of 2 uses ignore the return value.

Reply:
([See above comment](https://github.com/mongodb/mongo-java-driver/pull/1102#discussion_r1156328535))

Reply:
Done now